examples/src/java/org/pantsbuild/example/3rdparty_jvm.md

```diff
@@ -291,3 +291,237 @@ time pants is run (otherwise, pants will cache it):
             mutable=True),
       ],
     )
+
+Strict Dependencies
+-------------------
+
+### What is strict_deps?
+
+Strict_deps is a feature of JVM targets which controls the dependencies
+available on the compile classpath when that target is being compiled.
+
+When compiling a JVM target with strict_deps disabled, all transitive
+dependencies of that target are included on the compile classpath.
+
+When compiling a JVM target with strict_deps enabled, the only things on the
+compile classpath are those listed in its `dependencies` attribute and those
+listed in the `exports` attribute of any target's transitive `dependencies`.
+
+Enabling strict_deps speeds up builds because most targets don't need access
+to their full transitive dependencies, so 1) it reduces the work the compiler
+has to do when searching for symbols, and 2) it improves cache hit rates by
+reducing the number of targets which contribute to cache keys. However, it
+requires you to be more exact in writing down your dependencies.
```

Comment:
Maybe
> Enabling strict_deps speeds up builds because it 1) reduces the work the compiler has to do when searching for symbols, and 2) improves cache hit rates by reducing the number of dependencies that contribute to a targets cache keys. This works because most targets don't need their full transitive dependencies available to compile. However, it does require you to be more exact in writing down your dependencies.